lib/api/multiObjectDelete.js

```diff
@@ -273,26 +267,22 @@ export function getObjMetadataAndDelete(authInfo, canonicalID, request,
             // delete markers; Utapi must be updated to allow this for
             // 'deleteObject' metrics
             let isDeleteMarker;
-            let deleteMarker;
-            // If trying to delete an object that does not exist (empty objMD)
-            // return DeleteMarkerVersionId equals the versionID generated and
-            // DeleteMarker equals true
-            if (!objMD) {
+            let deleteMarkerVersionId;
+            // If trying to delete an object that does not exist (if a new
```

Comment:
we also create delete markers when an object exists but no version was specified, and versioning has been configured